plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/edit.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { getSettingWithCoercion } from '@woocommerce/settings';
+import { isBoolean } from '@woocommerce/types';
+
+/**
+ * Internal dependencies
+ */
+import { useIsDescendentOfSingleProductBlock } from '../../atomic/blocks/product-elements/shared/use-is-descendent-of-single-product-block';
+import { AddToCartOptionsSettings } from './settings';
+import { INNER_BLOCKS_TEMPLATE } from './constants';
+export interface Attributes {
+	className?: string;
+	isDescendentOfSingleProductBlock: boolean;
+}
+
+export type FeaturesKeys = 'isBlockifiedAddToCart';
+
+export type FeaturesProps = {
+	[ key in FeaturesKeys ]?: boolean;
+};
+
+export type UpdateFeaturesType = ( key: FeaturesKeys, value: boolean ) => void;
+
+// Pick the value of the "blockify add to cart flag"
+const isBlockifiedAddToCart = getSettingWithCoercion(
```

Comment:
Do we need this? My assumption is that if this flag is false, this block shouldn't be registered at all. So at the point we arrive at this code path, `isBlockifiedAddToCart` would always be true, no?

Reply:
Done in [4f9edfb](https://github.com/woocommerce/woocommerce/pull/53215/commits/4f9edfb92f3968a7c0c14b01e2313638a1c992b1)